austation/code/modules/popes-xenoarchaeology/xenoartifact_console.dm

```diff
@@ -0,0 +1,271 @@
+                sold_artifacts += list(info)
+                say(info)
+                qdel(I)
+                addtimer(CALLBACK(src, .proc/generate_new_buyer), (rand(1,5)*60) SECONDS)
+                return
+            else if(istype(I, /obj/structure/xenoartifact))
+                var/obj/structure/xenoartifact/X = I
+                final_price = X.modifier*X.price
+                if(final_price < 0)
+                    final_price = X.price*0.1
+                budget.adjust_money(final_price)
+                linked_techweb.add_point_type(TECHWEB_POINT_TYPE_DEFAULT, final_price*10)
+                info = "[X.name] sold at [station_time_timestamp()] for [final_price] credits, bought for [X.price]"
+                sold_artifacts += list(info)
+                say(info)
+                qdel(I)
+                addtimer(CALLBACK(src, .proc/generate_new_buyer), (rand(1,5)*60) SECONDS)
+                return
+            else    
+                final_price = 125*rand(0.3, 1.8) //This may be a point of conflict/balance
+                info = "[I] sold at [station_time_timestamp()] for [final_price]. No further information available."
+                sold_artifacts += list(info)
+                say(info)
+                qdel(I)
+                addtimer(CALLBACK(src, .proc/generate_new_buyer), (rand(1,5)*60) SECONDS)
+                return
+
+    update_icon()
+
+/obj/machinery/computer/xenoartifact_console/proc/generate_new_seller()
+    var/datum/xenoartifactseller/S = new
+    S.generate()
+    sellers += S
+
+/obj/machinery/computer/xenoartifact_console/proc/generate_new_buyer()
+    var/datum/xenoartifactseller/buyer/B = new
+    B.generate()
+    buyers += B
+    addtimer(CALLBACK(src, .proc/qdel, B), (rand(1,5)*60) SECONDS)
+
+/obj/machinery/computer/xenoartifact_console/proc/sync_devices()
+    for(var/obj/machinery/xenoartifact_inbox/I in oview(3,src))
+        if(I.linked_console != null || I.panel_open)
+            return
+        if(!(linked_inbox))
+            linked_inbox = I
+            linked_machines += list(I.name)
+            I.linked_console = src
+            say("Successfully linked [I].")
+            return
+    say("Unable to find linkable hadrware.")
+
+/obj/machinery/xenoartifact_inbox
+    name = "bluespace straythread pad" //Science words
+    desc = "This machine takes advantage of bluespace thread manipulation to highjack in-coming and out-going bluespace signals. Science uses it to deliver their very legal purchases." //All very sciencey
+    icon = 'icons/obj/telescience.dmi'
+    icon_state = "qpad-idle"
+    circuit = /obj/item/circuitboard/machine/xenoartifact_inbox
+    var/linked_console
+
+/obj/machinery/xenoartifact_inbox/proc/get_item_prox(var/dist) //Returns a list of items & strucutres, name is funked
+    var/obj/items = list()
+    for(var/obj/I in oview(dist, src))
+        items += list(I)
+    return items
```

Comment:
what's the point of this proc? it serves the same purpose as just a plain `oview(dist, src))`

Reply:
I think it originally had more content but I cut it down to this and just forgot to remove it.